### PR TITLE
Refactor FXIOS-5286 [v109] Remove BVC dependence in TabManager

### PR DESCRIPTION
**Pull request by nishant2718**
# [FXIOS-5286](https://mozilla-hub.atlassian.net/browse/FXIOS-5286)

TLDR: Remove BVC dependence in TabManager using AppSessionManager. It was used to provide the startAtHome func with information about whether the app was opened from an external source.

## Related
See https://github.com/mozilla-mobile/firefox-ios/pull/12518 for more context. That PR introduced AppSessionManager.

### CODE REVIEW

**Comment by nishant2718** [comment]
previously run with success 

image

The only update since then are swiftLint fixes.

**Comment by nishant2718** [comment]
UPDATE

Removed the activationState check, since startAtHome can be set to `always`, and not just `afterFourHours`.

**Comment by nishant2718** [comment]
Passed once, failed on an unrelated smokeTest the second time (`testClearPrivateDataButtonDisabled`) but runs fine locally. Skipping BR run for the last commit.

**Review by OrlaM — APPROVED** [review]

File: Client/Frontend/Browser/StartAtHomeHelper.swift [hunk 1]
```diff
@@ -9,17 +9,22 @@ class StartAtHomeHelper: FeatureFlaggable {
     // Override only for UI tests to test `shouldSkipStartHome` logic
     private var isRunningUITest: Bool
     lazy var startAtHomeSetting: StartAtHomeSetting? = featureFlags.getCustomState(for: .startAtHome)
+    var launchSessionProvider: LaunchSessionProviderProtocol
 
-    init(isRestoringTabs: Bool,
-         isRunningUITest: Bool = AppConstants.isRunningUITests) {
+    init(appSessionManager: AppSessionProvider = AppContainer.shared.resolve(),
```
Comment (yoanarios) [inline Client/Frontend/Browser/StartAtHomeHelper.swift:14]: small nit: Most of the time in the project and me personally prefer to group the parameters with default values at the end of the list of parameters. I don't think we have discussed with the team so you can leave as it is, I just wanted to point out

File: Tests/ClientTests/Mocks/MockAppSessionManager.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+@testable import Client
+
+class MockAppSessionManager: AppSessionProvider {
```
Comment (yoanarios) [inline Tests/ClientTests/Mocks/MockAppSessionManager.swift:8]: ❤️

File: Tests/ClientTests/StartAtHomeHelperTests.swift [hunk 3]
```diff
@@ -43,6 +44,16 @@ class StartAtHomeHelperTests: XCTestCase {
         XCTAssertTrue(shouldSkip, "Expected to skip because is restoring tabs")
     }
 
+    func test_shouldSkipStartAtHome_openedFromExternalSource() {
```
Comment (yoanarios) [inline Tests/ClientTests/StartAtHomeHelperTests.swift:47]: ❤️

**Review by yoanarios — APPROVED** [review]

**Pull request merged**